Update README.md (#393)
DAGWorks-Inc/hamilton: README.md

Hamilton is a novel paradigm for specifying a flow of delayed execution in python. It was originally built to simplify the creation of wide (1000+) column dataframes, but works on python objects of any type and dataflows of any complexity. Core to the design of Hamilton is a clear mapping of function name to components of the generated artifact, allowing you to quickly grok the relationship between the code you write and the data you produce. This paradigm makes modifications easy to build and track, ensures code is self-documenting, and makes it natural to unit test your data transformations. When connected together, these functions form a [Directed Acyclic Graph](https://en.wikipedia.org/wiki/Directed_acyclic_graph) (DAG), which the Hamilton framework can execute, optimize, and report on.

# Feature comparison

## Hamilton Does

> - Help you express your dataflow (data, ML, LLM, microservice) logic in a readable, self-documenting way in python

Reviewer: ```suggestion
- Help you express your dataflow (data, ML, LLM, micro-service) logic in a readable, self-documenting way in python.
```

> - Enable you to break your code into small, modular, and unit-testable pieces

Reviewer: ```suggestion
- Enable you to break your code into small, modular, and unit-testable pieces.
```

> - Make it easier to port these flows between different contexts

Reviewer: ```suggestion
- Make it easier to port dataflows between different execution contexts.
```

> - Scale as much as any underlying library you might choose to use

Reviewer: ```suggestion
- Scale as much as any underlying library you might choose to use.
```

> - Work nicely with a variety of computational/orchestration tools

Reviewer: ```suggestion
- Work nicely with a variety of computational/orchestration tools.
```

## Hamilton Does Not

> - Provision infrastructure -- an orchestration/execution framework is what you're looking for, and Hamilton will happily run on top

Reviewer: ```suggestion
- Provision infrastructure -- a macro orchestration/execution framework is what you're looking for. Hamilton does have hooks/integrations for a few frameworks.
```

> - Care about the modeling logic you use

Reviewer: ```suggestion
- Care about what you put inside a function, or what ML modeling/methodology you use.
```

> - Store execution data/telemetry on your behalf (see the [dagworks product](www.dagworks.io)

Reviewer: ```suggestion
- Come with storing execution data/telemetry on your behalf. If that is of interest, see the [DAGWorks product](www.dagworks.io).
```

> - Work with non-python logic (aside from tooling such as SQL, which can happily interfact with python tooling)

Reviewer: ```suggestion
- Work with non-python logic (aside from tooling such as SQL, which can happily interfact with python tooling).
```

> See the table below for more specifics/how it compares to other common tooling:

Reviewer: ```suggestion
See the table below for more specifics/how it compares to other common tooling.
```
